src/components/utils/ComboBoxItems/ComboBoxItems.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { GetItemPropsOptions } from 'downshift';
+import Text from '../../Text';
+import Box from '../../Box';
+import MenuItem from '../MenuItem';
+import groupBy from 'lodash/groupBy';
+
+interface ComboBoxItemsProps<T> {
+  items: T[];
+  disableItem: (item: T) => boolean;
+  getItemProps: (options: GetItemPropsOptions<T>) => T;
+  itemToString: (item: T) => string;
+  itemToGroup?: (item: T) => string | undefined;
+  selectedItem?: T | null;
+}
+
+/**
+ * Acts as a wrapper of MenuItems in a list of dropdown options,
+ * to allow grouping.
+ */
+function ComboBoxItems<Item>({
+  items,
+  itemToGroup = () => undefined,
+  ...rest
+}: ComboBoxItemsProps<Item>): React.ReactElement<ComboBoxItemsProps<Item>> {
+  const groupedItems = React.useMemo(() => {
+    return groupBy(items, itemToGroup);
+  }, [items, itemToGroup]);
+
+  return (
+    <>
+      {Object.entries(groupedItems).map(([groupName, items]) => {
+        if (groupName === 'undefined') {
```

Comment:
This is bug-prone and we shouldn't opt to do that. I think we should remove this ternary and just opt to either have no groups or to make all items **always** belong to a group. If they shouldn't belong to a group, then the user can add a "fallback group" like "Other"